backend/bittan/bittan/views/views.py

```diff
@@ -1,67 +1,228 @@
+                            chapter_event=chapter_event
+                        )
+                except IntegrityError as e:
+                    continue
+                break
+            else: 
+                logging.critical("Failed to generate a ticket external id. This should never happen.")
+                return Response(status=500) # Returns status internal server error. 
+    request.session["reserved_payment"] = payment.pk
+    return Response(status=status.HTTP_201_CREATED)
+
+
+class StartPaymentRequestSerializer(serializers.Serializer):
+    email_address = serializers.EmailField(max_length=255)
+
+
+@api_view(['POST'])
+def start_payment(request):
+    response_data: dict
+    
+
+    valid_ser = StartPaymentRequestSerializer(data=request.data)
+    if valid_ser.is_valid():
+        response_data = valid_ser.validated_data
+    else:
+        return Response(
+                "InvalidRequestData",
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+    payment_id = request.session.get("reserved_payment")
+    if payment_id == None:
+        return Response(
+                "InvalidSession",
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+    payment = Payment.objects.get(pk=payment_id)
+
+    if payment.payment_started:
+        return Response(
+                "AlreadyPaidPayment",
+                status=status.HTTP_403_FORBIDDEN
+            )
+
+
+    tickets = payment.ticket_set.all()
+
+    # This handles if session expires before payment starts and should be compatible with 
+    # functionality that allows for buying tickets to several different ChapterEvents at once. 
+    grouped_tickets = tickets.values_list("chapter_event", "chapter_event__total_seats").annotate(count=Count("chapter_event__pk"))
+    if payment.status != PaymentStatus.RESERVED:
+        for chapter_event, total_seats, count in grouped_tickets:
+            chapter_event = ChapterEvent.objects.get(pk=chapter_event)
+            if count > total_seats - chapter_event.alive_ticket_count:
+                payment.status = PaymentStatus.FAILED_EXPIRED_RESERVATION
+                return Response(
+                    "SessionExpired", 
+                    status=status.HTTP_408_REQUEST_TIMEOUT
+                )
+        payment.expires_at = timezone.now() + grouped_tickets.aggregate(Min("chapter_event__reservation_duration"))["chapter_event__reservation_duration__min"]
+        payment.status = PaymentStatus.RESERVED
+        payment.save()
+
+    payment.payment_started = True
+    payment.save()
+    payment.email = response_data["email_address"]
+    payment.save()
+
+    total_price = tickets.aggregate(Sum("ticket_type__price"))["ticket_type__price__sum"]
+    
+    swish = Swish.get_instance() # Detta hämtar swish instansen som är global över hela bittan. I den här kan saker anropas. 
+    
+    #payment_request: SwishPaymentRequest = swish.create_swish_payment(total_price, chapter_event.swish_message)
```

Comment:
I guess you commented this out because we don't have a unique chapter_event associated with each payment? Still, we should handle this more elegantly and actually use `chapter_event.swish_message` somehow.

Reply:
You guessed correctly. Now this matter has been discussed and the conclusion is that the functionality to book tickets for several different `ChapterEvents` at once is to be scrapped as of now. A new version that removes this functionality from this PR as well as resolves this has been made in 0fbd2d712091.